library/ctr_drbg.c

```diff
@@ -447,7 +458,7 @@ int mbedtls_ctr_drbg_random_with_add( void *p_rng,
 
     ctx->reseed_counter++;
```

Comment:
Are you sure you want to increment the reseed counter when `ret != 0`? I'm assuming `ret` can be non-zero here, since you wrote `return( ret );`.

`return( 0 )` made it obvious that reseed_counter was only incremented on success, because control flow would only reach here on success. Now I have to read the function more carefully to see if any error cases fall through to here.